plugins/flutter_aepedgeidentity/android/src/main/java/com/adobe/marketing/mobile/flutter/flutter_aepedgeidentity/FlutterAEPEdgeIdentityDataBridge.java

```diff
@@ -35,76 +33,75 @@ class FlutterAEPEdgeIdentityDataBridge {
 
-
-        if (map == null) {
+        if (map == null || map.isEmpty()){
             return null;
         }
 
-        if (map.isEmpty()) {
-            return new HashMap<>();
-        }
-
-        Map identityMapToMap = new HashMap<>();
+        Map identityMapToMap = new HashMap();
 
         for (String namespace : map.getNamespaces()) {
-            List<IdentityItem> items = map.getIdentityItemsForNamespace(namespace);
-            List itemsAsArray = new ArrayList();
+            List<IdentityItem> identityitems = map.getIdentityItemsForNamespace(namespace);
 
-            for (IdentityItem item : items) {
-                Map itemMap = new HashMap<>();
+            List<Map<String, Object>> identityItemsList = new ArrayList();
 
-                itemMap.put(IS_PRIMARY_KEY, item.isPrimary());
-                itemMap.put(AEP_AUTH_STATE_KEY, item.getAuthenticatedState().getName());
-                itemMap.put(ID_KEY, item.getId());
+            for (IdentityItem item : identityitems) {
+                Map identityItemAsMap = new HashMap();
 
-                itemsAsArray.add(itemMap);
+                identityItemAsMap.put(IS_PRIMARY_KEY, item.isPrimary());
+                identityItemAsMap.put(AEP_AUTH_STATE_KEY, item.getAuthenticatedState().getName());
+                identityItemAsMap.put(ID_KEY, item.getId());
+
+                identityItemsList.add(identityItemAsMap);
             }
 
-            if (itemsAsArray.size() != 0) {
-                identityMapToMap.put(namespace, itemsAsArray);
+            if (identityItemsList.size() != 0) {
+                identityMapToMap.put(namespace, identityItemsList);
             }
         }
         return identityMapToMap;
     }
 
-    static IdentityMap mapToIdentityMap(final Map<String, Map<String, List<Map<String, Object>>> >map) {
+    @SuppressLint("LongLogTag")
+    static IdentityMap mapToIdentityMap(final Map<String, Map<String, List<Map<String, Object>>>>map) {
 
-        if (map == null) {
+        if (map == null || map.isEmpty()) {
             return null;
         }
 
-        IdentityMap identityMapFromFlutterMap = new IdentityMap();
+        IdentityMap identityMap = new IdentityMap();
 
-        Map<String, List<Map<String, Object>>> mapvalue = map.get(IDENTITY_MAP_KEY);
+        Map<String, List<Map<String, Object>>> genericIdentityMap = map.get(IDENTITY_MAP_KEY);
 
-            Iterator<String> itr = mapvalue.keySet().iterator();
-            while (itr.hasNext()) {
-                List<Map<String, Object>> namespaceArray = null;
-                String namespace = itr.next();
-                try {
-                    namespaceArray = (ArrayList) mapvalue.get(namespace);
+        for (Map.Entry<String, List<Map<String, Object>>> entry : genericIdentityMap.entrySet()) {
 
-                } catch(ClassCastException e){
+            String namespace = entry.getKey();
 
-                }
+            List<Map<String, Object>> namespaceList = null;
```

Comment:
Use this 
```
List<Map<String, Object>> namespaceList;
 try {
      namespaceList = (ArrayList) entry.getValue();             
 } catch (ClassCastException e) {
        Log.d("FlutterAEPEdgeIdentityDataBridge", "mapToIdentityMap: " + e);
}
```

Reply:
You don't need try catch as it already a list. You can replace the entire logic with this.

`List<Map<String, Object>> namespaceList = entry.getValue(); `